b64filter does not halt when child command fails

I've been able to reproduce it with this command:
```
gzip -cd ~/hieu/nn/0/1/sentences.gz | b64filter parallel --halt 2 -j 16 --pipe -k -l 1024 false
```
`parallel` will indicate that `false` failed, but b64filter won't return and hang indefinitely.

It only happens for small sets of input sentences. If I try this with a sentences.gz from English, it will fail and stop on a pipe error.

```
go get -u github.com/paracrawl/b64filter
[cs-vand1@login-e-12 ~]$ gzip -cd ~/hieu/nn/0/1/sentences.gz | b64filter parallel --halt 2 -j 16 --pipe -k -l 1024 false
parallel: This job failed:
false
2020/03/11 10:04:15 b64filter.go:257: error waiting for command: exit status 1
```
👍

--- a/b64filter.go
+++ b/b64filter.go
@@ -254,6 +254,10 @@ func main() {
 	}
 	cmdin.Close()
 
+	if err = cmd.Wait(); err != nil {
+		log.Fatalf("error waiting for command: %v", err)
+	}
+
 	// wait for the queue to drain
 	for {
 		if counts.Empty() {
@@ -266,9 +270,6 @@ func main() {
 	// it is required that all reading from the command is done before
 	// calling Wait(). 
 	_ = <-done
-	if err = cmd.Wait(); err != nil {
-		log.Fatalf("error waiting for command: %v", err)
-	}
 
 	log.Printf("processed %v documents", i)
 }

